### ISSUE TITLE
MissingFile reported despite a file_exists() guard when the path contains ".."

### ISSUE THREAD

**Issue by Aerendir** [vimeo/psalm#11983]
A `file_exists()` (or `is_file()`) guard stops Psalm from reporting `MissingFile` on the `require` it protects — the behaviour introduced by #9012 for #3886. The guard is ignored when the path contains `..`, so the same code is reported or not depending only on how the path is written.

### Reproducer

psalm.dev cannot run it, because the playground rejects file includes (`File includes are not allowed per your Psalm config - check the allowFileIncludes flag`), so here it is as three files.

`psalm.xml`:

```xml
<?xml version="1.0"?>
<psalm errorLevel="1" findUnusedCode="false">
    <projectFiles>
        <directory name="tests"/>
    </projectFiles>
</psalm>
```

`tests/dotdot.php`:

```php
<?php
if (file_exists(__DIR__ . "/../optional.php")) {
    require __DIR__ . "/../optional.php";
}
```

`tests/dirname.php`:

```php
<?php
if (file_exists(dirname(__DIR__) . "/optional.php")) {
    require dirname(__DIR__) . "/optional.php";
}
```

`optional.php` does not exist. Running `vendor/bin/psalm --no-cache`:

```
tests/dotdot.php:3:5:error - MissingFile: Cannot find file /tmp/psalm/optional.php to include (see https://psalm.dev/107)
```

Both files guard the same file, but only the one whose path contains `..` is reported.

Reproduced on 6.13.1, 6.18.0 and `dev-master` (b131e78705e702c76030a6683f557276f08e5a51).

### Cause

The two sides of the guard store and look up the path in different forms:

- `file_exists()` records the path as written, e.g. `/tmp/psalm/tests/../optional.php` — [`NamedFunctionCallHandler.php#L205`](https://github.com/vimeo/psalm/blob/b131e78705e702c76030a6683f557276f08e5a51/src/Psalm/Internal/Analyzer/Statements/Expression/Call/NamedFunctionCallHandler.php#L205)
- `require` normalizes it first, to `/tmp/psalm/optional.php` — [`IncludeAnalyzer.php#L152`](https://github.com/vimeo/psalm/blob/b131e78705e702c76030a6683f557276f08e5a51/src/Psalm/Internal/Analyzer/Statements/Expression/IncludeAnalyzer.php#L152) — and then looks it up in `phantom_files` — [`IncludeAnalyzer.php#L251`](https://github.com/vimeo/psalm/blob/b131e78705e702c76030a6683f557276f08e5a51/src/Psalm/Internal/Analyzer/Statements/Expression/IncludeAnalyzer.php#L251)

Without `..` the two forms coincide, which is why `dirname(__DIR__)` works. Normalizing the path in `NamedFunctionCallHandler` before storing it in `phantom_files` would make the two agree.

### Why it matters

Rector's `DirnameDirConcatStringToDirectStringPathRector` rewrites `dirname(__DIR__) . '/x.php'` into `__DIR__ . '/../x.php'`. The rule was introduced in [rectorphp/rector-src#7552](https://github.com/rectorphp/rector-src/pull/7552) and added to the `CODE_QUALITY` set in [rectorphp/rector-src#8470](https://github.com/rectorphp/rector-src/pull/8470), first released in Rector 2.6.7. Since then, in projects that use that set, a guarded include written with `dirname(__DIR__)` gets rewritten into the form Psalm reports: code Psalm accepted starts failing after an automated refactoring that does not change behaviour. That is how we found it.
